alien4cloud/alien4cloud.go

```diff
@@ -0,0 +1,2169 @@
+func (c *a4cClient) UpdateCapabilityProperty(a4cCtx *TopologyEditorContext, componentName string, propertyName string, propertyValue string, capabilityName string) error {
+
+	if a4cCtx == nil {
+		return errors.New("Context object must be defined")
+	}
+
+	topoEditorExecute := TopologyEditorUpdateCapabilityProperty{
+		TopologyEditorExecuteRequest: TopologyEditorExecuteRequest{
+			NodeName:            componentName,
+			PreviousOperationID: a4cCtx.PreviousOperationID,
+			OperationType:       a4cUpdateCapabilityPropertyValueOperationJavaClassName,
+		},
+		PropertyName:   propertyName,
+		PropertyValue:  propertyValue,
+		CapabilityName: capabilityName,
+	}
+
+	if a4cCtx.TopologyID == "" {
+		var err error
+		a4cCtx.TopologyID, err = c.getA4CTopologyID(a4cCtx.AppID, a4cCtx.EnvID)
+		if err != nil {
+			return errors.Wrapf(err, "Unable to get A4C application topology for app %s and env %s", a4cCtx.AppID, a4cCtx.EnvID)
+		}
+	}
+
+	err := c.editA4CTopology(a4cCtx, topoEditorExecute)
+
+	if err != nil {
+		return errors.Wrapf(err, "Unable to edit the topology of application '%s' and environment '%s'", a4cCtx.AppID, a4cCtx.EnvID)
+	}
+
+	return nil
+}
+
+// AddNodeInA4CTopology Add a new node in the A4C topology
+// not used any more
+func (c *a4cClient) AddNodeInA4CTopology(a4cCtx *TopologyEditorContext, NodeTypeID string, nodeName string) error {
+
+	if a4cCtx == nil {
+		return errors.New("Context object must be defined")
+	}
+
+	a4cTopology, err := c.getA4CTopology(a4cCtx.AppID, a4cCtx.EnvID)
+
+	if err != nil {
+		return errors.Wrapf(err, "Unable to get A4C application topology for app %s and env %s", a4cCtx.AppID, a4cCtx.EnvID)
+	}
+
+	var nodeTypeVersion string
+
+	for _, node := range a4cTopology.Data.NodeTypes {
+		if NodeTypeID == node.ElementID {
+			nodeTypeVersion = node.ArchiveVersion
+		}
+	}
+
+	if reflect.DeepEqual(nodeTypeVersion, reflect.Zero(reflect.TypeOf(nodeTypeVersion)).Interface()) {
+		return errors.Wrapf(err, "Unable to get archive version for node '%s' from A4C application topology for app %s and env %s", NodeTypeID, a4cCtx.AppID, a4cCtx.EnvID)
+	}
+
+	topoEditorExecute := TopologyEditorAddNode{
+		TopologyEditorExecuteRequest: TopologyEditorExecuteRequest{
+			NodeName:            nodeName,
+			PreviousOperationID: a4cCtx.PreviousOperationID,
+			OperationType:       a4cAddNodeOperationJavaClassName,
+		},
+		NodeTypeID: NodeTypeID + ":" + nodeTypeVersion,
+	}
+
+	if a4cCtx.TopologyID == "" {
+		a4cCtx.TopologyID, err = c.getA4CTopologyID(a4cCtx.AppID, a4cCtx.EnvID)
+		if err != nil {
+			return errors.Wrapf(err, "Unable to get A4C application topology for app %s and env %s", a4cCtx.AppID, a4cCtx.EnvID)
+		}
+	}
+
+	err = c.editA4CTopology(a4cCtx, topoEditorExecute)
+
+	if err != nil {
+		return errors.Wrapf(err, "Unable to edit the topology of application '%s' and environment '%s'", a4cCtx.AppID, a4cCtx.EnvID)
+	}
+
+	return nil
+}
+
+// AddRelationship Add a new relationship in the A4C topology
+// not used any more
```

Comment:
Remove either the comment or the function

Reply:
removed the comment